### DISCUSSION TITLE
The order of spawnOnOnce

### DISCUSSION THREAD

**Discussion by refaelsh** [xmonad/xmonad#485] [Q&A] [answered]
I have something like the following in my config:
```haskell
spawnOnOnce "1" "firefox"
spawnOnOnce "2" $ "neovide /home/refaelsh/repos/giter/"
spawnOnOnce "3" $ myTerminal ++ " -d=~/repos/giter"
spawnOnOnce "9" "signal-desktop"
spawnOnOnce "0" $ "cd ~/repos/dotfiles/ && neovide configuration.norg"
```

But, I get the following results:
1. Firefox is on 2 instead of on 1.
1. What should have been on 2 shows up on 1.
1. What should have been on 0 shows up also on 1.

What am I doing wrong, please?

**Comment by geekosaur** [comment]
First, do you have `manageSpawn` in the `manageHook`? Without that, things won't be moved.
Second, I've found browsers to move themselves around, switch workspaces, and do other annoying things on their own.

**Reply by refaelsh** [reply]
> First, do you have manageSpawn in the manageHook?

Yes, I do.

> Second, I've found browsers to move themselves around, switch workspaces, and do other annoying things on their own.

Interesting. What so special about browsers?

**Comment by geekosaur** [comment]
They think they own the whole desktop :(

**Reply by refaelsh** [reply]
:-)

**Comment by portnov** [comment]
Firefox is also known for changing it's WM_CLASS after window is created. Maybe other atoms are changed after creation too.

**Reply by refaelsh** [reply]
Damn,,, This sounds bad. I wonder what was the technical decision for this kind of thing.
Any way, that would explain Firefox, but what about Neovide?

**Comment by geekosaur** [comment, marked as answer]
In the case of Firefox, it's because it doesn't know what to assign to `WM_CLASS` until the window's Javascript runs.
